Source/SwiftLintFramework/Rules/ObjectLiteralRule.swift

```diff
@@ -0,0 +1,115 @@
+            "let color = NSColor(red: value, green: value, blue: value, alpha: 1)"
+        ],
+        triggeringExamples: ["", ".init"].flatMap { (method: String) -> [String] in
+            ["UI", "NS"].flatMap { (prefix: String) -> [String] in
+                [
+                    "let image = ↓\(prefix)Image\(method)(named: \"foo\")",
+                    "let color = ↓\(prefix)Color\(method)(red: 0.3, green: 0.3, blue: 0.3, alpha: 1)",
+                    "let color = ↓\(prefix)Color\(method)(red: 100 / 255.0, green: 50 / 255.0, blue: 0, alpha: 1)",
+                    "let color = ↓\(prefix)Color\(method)(white: 0.5, alpha: 1)"
+                ]
+            }
+        }
+    )
+
+    public func validateFile(_ file: File, kind: SwiftExpressionKind,
+                             dictionary: [String : SourceKitRepresentable]) -> [StyleViolation] {
+        guard kind == .call,
+            let offset = (dictionary["key.offset"] as? Int64).flatMap({ Int($0) }),
+            isImageNamedInit(dictionary, file: file) || isColorInit(dictionary, file: file) else {
+            return []
+        }
+
+        return [
+            StyleViolation(ruleDescription: type(of: self).description,
+                           severity: configuration.severity,
+                           location: Location(file: file, byteOffset: offset))
+        ]
+    }
+
+    private func isImageNamedInit(_ dictionary: [String : SourceKitRepresentable], file: File) -> Bool {
+        guard let name = dictionary["key.name"] as? String,
+            initsForClasses(["UIImage", "NSImage"]).contains(name),
+            case let arguments = dictionary.enclosedArguments,
+            arguments.flatMap({ $0["key.name"] as? String }) == ["named"],
+            let argument = arguments.first,
+            case let kinds = kindsFor(argument, file: file),
+            kinds == Set(arrayLiteral: .string) else {
+                return false
+        }
+
+        return true
+    }
+
+    private func isColorInit(_ dictionary: [String : SourceKitRepresentable], file: File) -> Bool {
+        guard let name = dictionary["key.name"] as? String,
+            initsForClasses(["UIColor", "NSColor"]).contains(name),
+            case let arguments = dictionary.enclosedArguments,
+            case let argumentsNames = arguments.flatMap({ $0["key.name"] as? String }),
+            argumentsNames == ["red", "green", "blue", "alpha"] || argumentsNames == ["white", "alpha"],
+            validateColorKinds(arguments: arguments, file: file) else {
+                return false
+        }
+
+        return true
+    }
+
+    private func initsForClasses(_ names: [String]) -> [String] {
+        return names.flatMap { name in
+            [
+                name,
+                name + ".init"
+            ]
+        }
+    }
+
+    private func validateColorKinds(arguments: [[String: SourceKitRepresentable]], file: File) -> Bool {
+        for dictionary in arguments {
```

Comment:
`[.number]` in my example is actually a `Set<SyntaxKind>`! Not an `Array<SyntaxKind>`. Because `Set` conforms to `ArrayLiteralConvertible`, it can _literally_ be converted from an an array literal.